src/Paket.PowerShell/CmdletExt.fs

```diff
@@ -37,4 +40,17 @@ type PSCmdlet with
             else false
 
     member x.SetCurrentDirectoryToLocation() =
-        System.Environment.CurrentDirectory <- x.SessionState.Path.CurrentFileSystemLocation.Path
\ No newline at end of file
+        Environment.CurrentDirectory <- x.SessionState.Path.CurrentFileSystemLocation.Path
+
+    member x.RegisterTrace() =
+        Logging.verbose <- x.Verbose
+        let id = Threading.Thread.CurrentThread.ManagedThreadId
+        Logging.subscribe (fun trace ->
+            if id = Threading.Thread.CurrentThread.ManagedThreadId then
```

Comment:
This thread ID check appears it will work. Messages in `async` will not print, but also will not error out the program now.